### ISSUE TITLE
Counter accepts a double, but StatsdLineBuilder loses precision by converting to Long.

### ISSUE THREAD

**Issue by tyrantkhan** [micrometer-metrics/micrometer#2512]
Hi,

I have a use-case where I am incrementing a counter by a decimal value. I'm also using the statsdregistry.

```
@Override
    public void increment(double amount) {
        if (!shutdown && amount > 0) {
            count.add(amount);
            sink.next(lineBuilder.count((long) amount));
        }
    }
```

While, I'm not sure where the count is being used, but by having StatsdLineBuilder.count() accept a long, the following happens:
```
@Override
    public String count(long amount, Statistic stat) {
        return line(Long.toString(amount), stat, TYPE_COUNT);
    }
```

It seems like the default implementation is to convert to a string anyways, I'm hoping we can update the StatsdLineBuilder interface to accept a string or double.

**Comment by shakuzen** [comment]
We can probably make this possible. It's not clear to me whether the StatsD line format specifies decimal numbers as acceptable for counters. It may depend on the specific implementation. Any documentation I've found does not explicitly say the metric value should be integer or not but all examples seem to be with integers. I did try running the etsy statsd daemon and sending the following which seemed to work:

```
foo:1.1|c
```

So I'm not sure if we're safe to allow decimal counter values in general or not. Looking at other libraries, [tim-group/java-statsd-client](https://github.com/tim-group/java-statsd-client/blob/ec75702a61af3daabd9c826f687243a3ee6fcde9/src/main/java/com/timgroup/statsd/StatsDClient.java#L27-L52) does not accept decimals for counters, but the datadog [java-dogstatsd-client](https://github.com/DataDog/java-dogstatsd-client/blob/a1272afbcb433483e2f9f9537a9e8061d78dc5db/src/main/java/com/timgroup/statsd/StatsDClient.java#L70-L102) does accept decimal values.
